test/system/components/student/tracks_list_test.rb

```diff
@@ -3,54 +3,56 @@
 module Components
   module Student
     class TracksListTest < ApplicationSystemTestCase
-      include TableMatchers
-
       def setup
         super
 
-        # This component uses the API, which requires authentication
+        # This component uses the API, which requires authentication.
         sign_in!
       end
 
       test "shows tracks" do
-        track = create :track, title: "Ruby"
-        concept_exercises = Array.new(10).map { create :concept_exercise, track: track }
-        practice_exercises = Array.new(10).map { create :practice_exercise, track: track }
+        track = create :track,
+          title: "Ruby",
+          tags: ["Foo:Bar", "Abc:Xyz"]
+        concept_exercises = Array.new(3).map { create :concept_exercise, track: track }
+        practice_exercises = Array.new(4).map { create :practice_exercise, track: track }
         create :user_track, track: track, user: @current_user
-        concept_exercises.sample(5).map { |ex| create :concept_solution, user: @current_user, exercise: ex }
-        practice_exercises.sample(5).map { |ex| create :practice_solution, user: @current_user, exercise: ex }
+        concept_exercises.sample(1).map { |ex| create :concept_solution, user: @current_user, exercise: ex }
+        practice_exercises.sample(2).map { |ex| create :practice_solution, user: @current_user, exercise: ex }
 
         visit test_components_student_tracks_list_url
 
-        row = {
-          "Icon" => lambda {
-            assert_css "img[src='https://assets.exercism.io/tracks/ruby-hex-white.png'][alt='icon for Ruby track']"
-          },
-          "Title" => "Ruby",
-          "Num concept exercises" => 10,
-          "Num practice exercises" => 10,
-          "URL" => "https://test.exercism.io/tracks/ruby",
-          "New?" => "true",
-          "Tags" => "OOP, Web Dev",
-          "Joined?" => "true",
-          "Num completed concept exercises" => 5,
-          "Num completed practice exercises" => 5,
-          "Progress" => "Completed concept exercises: 25%, Uncompleted concept exercises: 25%, "\
-                        "Completed practice exercises: 25%, Uncompleted practice exercises: 25%"
-        }
-        assert_table_row first("table"), row
+        assert_html '
```

Comment:
I don't know how to feel about this. I feel like it may be brittle, but at the same time I like it because it's accurate. Every time we change the components, we might really want to change the system test. I want to see where this approach takes us.

Reply:
I'm also not sure. Is the goal to verify the exact HTML, or that the right information is displayed? I have no idea how system tests are usually used in Ruby though. @SleeplessByte?

Reply:
> Every time we change the components, we might really want to change the system test

Yeah, so this was the conclusion I came to.

>  Is the goal to verify the exact HTM

And yes, I think it is.

---

I think what this gives us is a very firm test that our components are rendering correctly, which protects us from inadvertently changing something, or React etc changing which changes something unexpectedly. Having the firmness of this test here then means I would happily just assert `.c-track` exists anywhere else in the tests, knowing that we have this guarantee that the track component renders properly.

That was my thinking anyway. I spent a lot of time wondering whether this was good or stupid, but settled on good 🙂

Reply:
I'm not a fan of this either, but YMMV. 

It's fine tho; just go with it.

Reply:
So three non-fans, but none of us feel that strongly about it :)